js/connection.js

```diff
@@ -151,21 +151,33 @@ weechat.factory('connection',
                     _requestHotlist().then(function(hotlist) {
                         handlers.handleHotlistInfo(hotlist);
 
-                        if (successCallback) {
-                            successCallback();
-                        }
                     });
+                    // Schedule hotlist syncing every so often so that this
+                    // client will have unread counts (mostly) in sync with
+                    // other clients or terminal usage directly.
+                    setInterval(function() {
+                        if ($rootScope.connected) {
+                            _requestHotlist().then(function(hotlist) {
```

Comment:
I'm not sure that is needed or correct always. If it is, we should send hotlistclear for every recieved line that is in an open buffer. I think hotlistclear should always only be sent for user actions.